File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @param {number} k
     * @return {number}
     */
    kTarget = null
    counter = 0
    result = null

    kthSmallest(root, k) {
        this.kTarget = k
        this.dfs(root)
        return this.result
    }

    dfs(root) {
        if(!root || this.result) return

        this.dfs(root.left)
        if(this.result) return
        this.counter++
        if(this.counter == this.kTarget) {
            this.result = root.val
            return
        }
        this.dfs(root.right)
    }

}

File: Data Structures & Algorithms/sqrtx/submission-1.py
class Solution:
    def mySqrt(self, x: int) -> int:
        left, right = 0, x
        while left <= right:
            mid = left + (right - left) // 2
            sq = mid * mid
            if sq == x:
                return mid
            if sq < x:
                left = mid + 1
            elif sq > x:
                right = mid - 1
        return right